core/src/main/java/org/elasticsearch/action/fieldstats/TransportFieldStatsTransportAction.java

```diff
@@ -208,13 +212,19 @@ protected FieldStatsShardResponse shardOperation(FieldStatsShardRequest request)
                         // ignore.
                         continue;
                     }
-                    FieldStats<?> stats = fieldType.stats(searcher.reader());
-                    if (stats != null) {
-                        fieldStats.put(matchField, stats);
+                    BytesReference cacheKey = new BytesArray("fieldstats:" + matchField);
```

Comment:
general note about this class, I think we should deduplicate the field names first and then fetch otherwise we might do it X times?